Swift Sources/UIViewController+Promise.swift

```diff
@@ -19,6 +20,15 @@ class MFMailComposeViewControllerProxy: NSObject, MFMailComposeViewControllerDel
         }
         PMKRelease(self)
     }
+
+    func messageComposeViewController(controller: MFMessageComposeViewController!, didFinishWithResult result: MessageComposeResult) {
+        if result.value == MessageComposeResultFailed.value {
+            controller.reject(NSError(domain: PMKErrorDomain, code: 4, userInfo: [NSLocalizedDescriptionKey: "The user’s attempt to save or send the message was unsuccessful."]))
```

Comment:
Alright so what are we doing with the error codes? Right now they don't match at all between Swift and ObjC. Are we going to try to match them up? Or just add a PMKOperationFailed code to both and be done?